### PR TITLE
Remove istanbul.v1 code

### PR DESCRIPTION
**Pull request by hbandura**
Remove all code and references of the deprecated istanbul v1 consensus code. The only remnant is the v1 PreparedCertificate since it's still used in storage.

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit 0fdd41b8e5c99b1bd1b9b987323b0e305fa836a3
<details><summary>coverage:  51.1% of statements across all listed packages</summary><br>
<pre>
coverage:  63.2% of statements in consensus/istanbul
coverage:  42.7% of statements in consensus/istanbul/announce
coverage:  55.9% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  24.3% of statements in consensus/istanbul/backend/internal/replica
coverage:  67.2% of statements in consensus/istanbul/core
coverage:  50.0% of statements in consensus/istanbul/db
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  64.4% of statements in consensus/istanbul/uptime
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: 3146592755
</details>

**Comment by palango** [comment]
@hbandura Have you thought about removing all `V2` suffixes as well?

File: e2e_test/e2e_test.go [hunk 1]
```diff
@@ -24,14 +24,9 @@ import (
...
-	println("====== RUNNING e2e tests with original consensus protocol")
-	code1 := m.Run()
-
-	// Activate V2 consensus block
-	test.BaseEthConfig.Istanbul.V2Block = big.NewInt(0)
 	println("====== RUNNING e2e tests with v2 consensus protocol")
 	code2 := m.Run()
-	os.Exit(code1 | code2)
+	os.Exit(code2)
 }
```
Comment (piersy) [inline e2e_test/e2e_test.go:30]: We can just remove this whole function.

**Comment by palango** [comment]
> I have, but since we are already retaining the old msg codes (though deprecating them), I thought of just leaving the 'v2', making it explicit that this is a new version of the consensus code.

What do you mean with retaining the old message codes?

> If we were to remove that suffix, I'd still do it in a different PR since I'd like to avoid the mess in the pr diff.

Sounds reasonable to me.

**Comment by hbandura** [comment]
> > I have, but since we are already retaining the old msg codes (though deprecating them), I thought of just leaving the 'v2', making it explicit that this is a new version of the consensus code.
> 
> What do you mean with retaining the old message codes?

The istanbul msg codes, PreprepareV1 and RoundChangeV1 . I left them with a prepended 'deprecated' in the name. 

I'm not against removing all references to v2, but I'd rather be sure that at least all validators are running versions that already removed all v1 code.

File: consensus/istanbul/core/roundchange_v2.go [hunk 2]
```diff
@@ -238,7 +223,7 @@ func (c *core) handleRoundChangeV2(msg *istanbul.Message) error {
 // ----------------------------------------------------------------------------
 
 // CurrentRoundChangeSet returns the current round change set summary.
-func (c *core) CurrentRoundChangeSetV2() *RoundChangeSetSummary {
+func (c *core) CurrentRoundChangeSet() *RoundChangeSetSummary {
```
Comment (gastonponti) [inline consensus/istanbul/core/roundchange_v2.go:226]: You've been maintaining the V2 suffix in all the PR except this one. Just checking if that was on purpose or not

**Review by gastonponti — APPROVED** [review]
Check the question but other than that, LGTM

**Pull request merged**